projects/graph/src/draw.py

```diff
@@ -1,14 +1,240 @@
+# """
+# General drawing methods for graphs using Bokeh.
+# """
+# import math
+
+# from bokeh.io import show, output_file
+# from bokeh.plotting import figure
+# from bokeh.models import (
+#     GraphRenderer,
+#     StaticLayoutProvider,
+#     Oval,
+#     LabelSet,
+#     ColumnDataSource,
+# )
+
+# # from bokeh.palettes import Spectral4
+# from graph import Graph
+
+
+# class BokehGraph:
+#     """Class that takes a graph and exposes drawing methods."""
+
+#     def __init__(self, graph):
+#         self.graph = graph
+#         self.keys = []
+#         self.values = []
+#         """
+#         {
+#             '0': {'1', '3'},
+#             '1': {'0'},
+#             '2': set(),
+#             '3': {'0'}
+#         }
+#         """
+#         for key in self.graph.vertices:
+#             for edge in self.graph.vertices[key]:
+#                 self.keys.append(key)
+#                 self.values.append(edge)
```

Comment:
I appreciate the fact that even though initially you struggled with this code, you still left your original work so you can learn from it. That shows me you take pride in your skills and you really want to learn and excel. 👍